pkg/controller/clusterdeprovision/clusterdeprovision_controller_test.go

```diff
@@ -195,13 +237,28 @@ func TestClusterDeprovisionReconcile(t *testing.T) {
 			}
 			existing := append(test.existing, test.deprovision, test.deployment)
 
-			fakeClient := fake.NewFakeClient(existing...)
+			mocks := setupDefaultMocks(t, existing...)
+
+			// This is necessary for the mocks to report failures like methods not being called an expected number of times.
+			defer mocks.mockCtrl.Finish()
+
+			if test.mockGetCallerIdentity {
+				mocks.mockAWSClient.EXPECT().
+					GetCallerIdentity(gomock.Any()).
+					Return(nil, test.expectedGetCallerIdentityError)
+			}
+
 			r := &ReconcileClusterDeprovision{
-				Client:               fakeClient,
+				Client:               mocks.fakeKubeClient,
 				scheme:               scheme.Scheme,
 				deprovisionsDisabled: test.deprovisionsDisabled,
 			}
 
+			// TODO: Mock this
+			actuators = []Actuator{&awsActuator{awsClientFn: func(clusterDeprovision *hivev1.ClusterDeprovision, c client.Client, logger log.FieldLogger) (awsclient.Client, error) {
```

Comment:
This is not safe to do in unit tests. It has side effects which leak out of the test.

Reply:
Fixed.

Reply:
This is still not going to allow for the unit tests to be run in parallel. But I am fine with letting it go as in. We can fix this and the hibernation controller later.

Reply:
Sounds good.